src/sessions/screen-share-session-handler.ts

```diff
@@ -47,39 +42,41 @@ export default class ScreenShareSessionHandler extends BaseSessionHandler {
   }
 
   async handleSessionInit (session: IJingleSession): Promise<void> {
-    await super.handleSessionInit(session);
-
-    session.on('terminated', (session: IJingleSession) => {
-      session._screenShareStream.getTracks().forEach((t: MediaStreamTrack) => t.stop());
-    });
+    try {
+      await super.handleSessionInit(session);
+    } catch (err) {
+      this.log(LogLevels.error, 'Screen share session init failed');
+      await super.endSession(session);
+    } finally {
+      const stream = await startDisplayMedia();
+
+      session.on('terminated', (session: IJingleSession) => {
+        session._screenShareStream.getTracks().forEach((t: MediaStreamTrack) => t.stop());
+      });
 
-    if (!this.sdk.isGuest) {
-      throwSdkError.call(this.sdk, SdkErrorTypes.not_supported, 'Screen share sessions not supported for authenticated users');
-    }
+      if (!this.sdk.isGuest) {
+        throwSdkError.call(this.sdk, SdkErrorTypes.not_supported, 'Screen share sessions not supported for authenticated users');
+      }
 
-    if (this.temporaryOutboundStream) {
-      this.temporaryOutboundStream.getTracks().forEach((track: MediaStreamTrack) => {
+      stream.getTracks().forEach((track: MediaStreamTrack) => {
         track.addEventListener('ended', this.onTrackEnd.bind(this, session));
       });
-      this.log(LogLevels.debug, 'temporaryOutboundStream exists. Adding stream to the session and setting it to _screenShareStream');
+      this.log(LogLevels.debug, 'Adding stream to the session and setting it to _screenShareStream');
```

Comment:
I know the old message didn't have this, but the pattern we try and follow for logging in the sdk is adding the details object with at least the conversationId and the sessionId if possible. This makes it so these logs get picked up in the conversationTimeline.